### PR TITLE
[WIN32K] Revert incorrect part of R50928 for RealizePalette.

### PR DESCRIPTION
**Pull request by Doug-Lyons**
## Revert incorrect code for Realize Palette

_Revert part of R50928 that causes Durak card suites to have wrong colors._

JIRA issue: [CORE-13748](https://jira.reactos.org/browse/CORE-13748) <= Durak Example
JIRA issue: [CORE-16510](https://jira.reactos.org/browse/CORE-16510) <= GDIProg Example

## Revert RealizePalette Code that causes wrong colors in Durak and GDIProg.

Durak in R50927
DURAK_Show_all_Cards-50927

Durak in R50928
DURAK_Show_all_Cards-50928

Correct GDIProg from CORE-17164
GDIProg_OK

GDIProg Menu Before:
GDIProg_Menu_Bad

GDIProg Menu After:
GDIProg_Menu_Good

GDIProg Palette Before:
GDIProg_Palette_Bad

GDIProg Palette After:
GDIProg_Palette_Good

### CODE REVIEW

**Review by JoachimHenze — APPROVED** [review]
happifies the bots and better real world results.
KVM: https://reactos.org/testman/compare.php?ids=77304,77308, LGTM
VBox: https://reactos.org/testman/compare.php?ids=77305,77309, LGTM

That justifies to merge this code, even when it obviously fingerpoints to "stuff remaining to be done in this context in the future".

File: win32ss/gdi/ntgdi/palette.c
Comment (binarymaster) [inline win32ss/gdi/ntgdi/palette.c:790]:
```suggestion
    DPRINT1("RealizePalette unimplemented for %s\n", (pdc->dctype == DC_TYPE_MEMORY ? "memory managed DCs" : "device DCs"));
```

File: win32ss/gdi/ntgdi/palette.c
Comment (binarymaster) [inline win32ss/gdi/ntgdi/palette.c:733]: You can just remove this commented out code.

File: win32ss/gdi/ntgdi/palette.c
Comment (binarymaster) [inline win32ss/gdi/ntgdi/palette.c:773]: Ditto.

**Comment by Doug-Lyons** [comment]
I have tested the https://jira.reactos.org/browse/CORE-17164 and the before is the same as the after when testing iDeaS, so I am removing the pictures from the description here that refer to iDeaS.

File: win32ss/gdi/ntgdi/palette.c
Comment (binarymaster) [inline win32ss/gdi/ntgdi/palette.c:773]: Does this debug print slow down the painting? If yes, better do `DPRINT` instead.

Comment (Doug-Lyons) [reply win32ss/gdi/ntgdi/palette.c:773]: From my past experience, this is not called that much, so leaving this as a DPRINT1 makes sense so that it will be logged and this can help us find why something is not working correctly. If it becomes a nuisance, then it can be demoted to a DPRINT later. This was a DPRINT1 in Revision 50927 before @zefklop added the Realize Palette code, so I prefer to leave it as it is now. Thanks.

**Comment by HBelusca** [comment]
Can you try some "graphical" DOS games (that use NTVDM) in ReactOS? I know NTVDM uses RealizePalette and I want to know whether the colors still work there.

**Comment by Doug-Lyons** [comment]
> Can you try some "graphical" DOS games (that use NTVDM) in ReactOS? I know NTVDM uses RealizePalette and I want to know whether the colors still work there.

Sorry, @HBelusca, I have never used computers for DOS games, so I don't even know where you would want me to start. If you want to provide me one or two names and links to their installs, I would be willing to test them for you though. To be able to do a color test, I would have to have them installed in Windows to compare as well, so this will take some work on my part.

File: win32ss/gdi/ntgdi/palette.c
Comment (zefklop) [inline win32ss/gdi/ntgdi/palette.c:778]: Does it change something if you disable this code only for memory DCs ?

Comment (Doug-Lyons) [reply win32ss/gdi/ntgdi/palette.c:778]: > I downloaded the full output from the testbot runs for both VBox and KVM and did a search for "RealizePalette unimplemented" which are the first two words in both of the DPRINT1's that were being used at that time to show memory DC's versus device DC's and there were zero (0) instances found. So this shows me that our regression tests do never exercise this code path. We have three test cases of which I am aware. These are 1) Durak, 2) GDIProg-Menu, and 3) GDIProg-Palette that use this code and these all use "memory DC's".

So I believe that disabling this code only for memory DC's would fix all known instances of problems with this code for now.

It would be good to have regression tests that exercise RealizePalette for both of these types of DC's, but I have no idea how to create them for now. Thanks.

Comment (Doug-Lyons) [reply win32ss/gdi/ntgdi/palette.c:778]: @zefklop, I did a search for RealizePalette in the regression tests and it look like you had started to create tests for this in 'modules\rostests\apitests\win32nt\ntuser\NtUserSelectPalette.c', but this code is #if 0'ed out. Maybe you have a better idea of how to finish these tests now? Thanks.

Comment (Doug-Lyons) [reply win32ss/gdi/ntgdi/palette.c:778]: I cleanup up this regression test code enough to get it to compile and when run it shows:
```
WARNING:  IntGdiRealizePalette at win32ss/gdi/ntgdi/palette.c:755 is UNIMPLEMENTED!
```
This is triggered when 'pdc->dctype == DCTYPE_DIRECT'
So I am not sure that we could ever reach the RealizePalette code with a "device DC".

**Comment by JoachimHenze** [comment]
Yes I do really like this PR's effect on the bots reducing crashes, and also on Duraks palette.
But I do highly recommend that we fix https://jira.reactos.org/browse/CORE-17626 first before we will merge PR3678 to master, to be able to retest the gdiprog case without that BSOD. I already tried: PR3678 will not help for that!

**Comment by Doug-Lyons** [comment]
@JoachimHenze, OK, Now that we can test the GDIProg program here https://jira.reactos.org/browse/CORE-17626 without a crash by changing its path are you comfortable with merging this to master? Thanks.

**Comment by Doug-Lyons** [comment]
Here are more DOS graphic game color comparisons that I created for MatterMost's Developer channel that does comparisons before and after this PR:

https://chat.reactos.org/reactos/pl/srr1soquxjnap8tx3ud6guwb3o

**Pull request merged**
